### ISSUE TITLE
Modes are filtered by origin mode even of no origin

### ISSUE THREAD

**Issue by timbl** [solid/acl-check#36]
Symptom:   Attempt to PUT an ACL file fails "Not all modes needed"  "missing mode: Control" when in fact the agent does have control.

This line seems to filter modes by origin trusted modes whether or not this request has an origin.

In https://github.com/solid/acl-check/blob/master/src/acl-check.js#L192

**Comment by michielbdejong** [comment]
If the origin is null then `originTrustedModes` should be all modes.
Where do you think it gets the filter list from? From the trusted apps for which origin?

**Comment by michielbdejong** [comment]
Right, I think this happens when you add your own domain as a trusted app, but with only some (not all) R/W/A/C modes.

If you don't have your same-origin domain as a trusted app, then `originTrustedModes.length === 0` and the filtering step will be skipped.
If you have your same-origin domain as a trusted app for all modes, then the filtering step will take place but have no effect.
If you have only some modes then this bug would surface.

That would happen for instance if a file is hosted on https://timbl.com but https://timbl.inrupt.net/profile/card#me (which includes the triple:
```turtle
[ n1:mode n1:Append, n1:Read, n1:Write; n1:origin <https://timbl.com> ]
```
 has control access to the file.

**Comment by timbl** [comment]
I have just spent a long time tracking a bug which turned out to be this one!!
Example:
- Origin rusted modes: Read, Write
- Origin? No
- Need: Control for creating .acl file

```
@prefix : <http://www.w3.org/ns/auth/acl#> .
    <#rwc>     a :Authorization;
         :accessTo <./>;
         :default <./>;
         :agent <https://www.w3.org/People/Berners-Lee/card#i>, 
                <https://timbl.inrupt.net/profile/card#me>;
         :mode :Control, :Read, :Write .
```
Here is a failing log:
```
    solid:ACL Using ACL https://example.com/example/.acl for ./2021/FooBar/.acl +13s
  solid:ACL    1  default authentications about <https://example.com/example/> in <https://example.com/example/.acl> +0ms
  solid:ACL accessDenied: checking access to <https://example.com/example/2021/FooBar/> by <https://example.inrupt.net/profile/card#me> and origin <https://example.com> +399ms
  solid:ACL    1  default authentications about <https://example.com/example/> in <https://example.com/example/.acl> +0ms
  solid:ACL Origin <https://example.com> is trusted +0ms
  solid:ACL   modesAllowed: Origin null is trusted. +0ms
  solid:ACL    Checking auth <https://example.com/example/.acl#rwc> with agent <https://example.inrupt.net/profile/card#me> +0ms
  solid:ACL     Agent explicitly authenticated. +0ms
  solid:ACL      Origin check not needed: no origin. +0ms
  solid:ACL       Mode allowed: <http://www.w3.org/ns/auth/acl#Read> +0ms
  solid:ACL       Mode allowed: <http://www.w3.org/ns/auth/acl#Write> +0ms
  solid:ACL accessDenied: modeURIorReasons: ["http://www.w3.org/ns/auth/acl#Read","http://www.w3.org/ns/auth/acl#Write"] +0ms
  solid:ACL  checking <http://www.w3.org/ns/auth/acl#Control> +0ms
  solid:ACL   MODE REQUIRED NOT ALLOWED: <http://www.w3.org/ns/auth/acl#Control> Denying with All Required Access Modes Not Granted +0ms

# eh?

  solid:ACL accessDenied: checking access to <https://example.com/example/> by <https://example.inrupt.net/profile/card#me> and origin <https://example.com> +0ms
  solid:ACL    1 direct authentications about <https://example.com/example/> +0ms
  solid:ACL Origin <https://example.com> is trusted +0ms
  solid:ACL   modesAllowed: Origin null is trusted. +0ms
  solid:ACL    Checking auth <https://example.com/example/.acl#rwc> with agent <https://example.inrupt.net/profile/card#me> +1ms
  solid:ACL     Agent explicitly authenticated. +0ms
  solid:ACL      Origin check not needed: no origin. +0ms
  solid:ACL       Mode allowed: <http://www.w3.org/ns/auth/acl#Read> +0ms
  solid:ACL       Mode allowed: <http://www.w3.org/ns/auth/acl#Write> +0ms
  solid:ACL accessDenied: modeURIorReasons: ["http://www.w3.org/ns/auth/acl#Read","http://www.w3.org/ns/auth/acl#Write"] +0ms
  solid:ACL  checking <http://www.w3.org/ns/auth/acl#Append> +0ms
  solid:ACL   Append required and Write allowed. OK +0ms


  solid:ACL Write access denied to https://example.inrupt.net/profile/card#me: 403 - All Required Access Modes Not Granted +0ms
  solid:server Error page because of: [HTTPError: All Required Access Modes Not Granted] { status: 403 } +13s
  solid:server Display no-permission for https://example.com/example/2021/FooBar/.acl +0ms


```

**Comment by timbl** [comment]
This was fixed, it seems, in commit e277ebad on 202101-31. by https://github.com/solid/acl-check/pull/37    Closing
